src/main/java/frc/robot/gamepads/Gamepad.java

```diff
@@ -0,0 +1,170 @@
+    private Button m_buttonRT;
+    private Button m_buttonSelect;
+    private Button m_buttonStart;
+
+    private final XboxControllerMapping m_xboxControllerMapping = new XboxControllerMapping();
+    private final LogitechControllerMapping m_logitechControllerMapping = new LogitechControllerMapping();
+
+    public Gamepad(int port, String controllerTitle) {
+        m_joystick = new Joystick(port);
+        m_controllerTitle = controllerTitle;
+
+        m_buttonA = createButton(ButtonType.A);
+        m_buttonB = createButton(ButtonType.B);
+        m_buttonX = createButton(ButtonType.X);
+        m_buttonY = createButton(ButtonType.Y);
+        m_buttonLB = createButton(ButtonType.LeftBumper);
+        m_buttonLT = createButton(ButtonType.LeftTrigger);
+        m_buttonRB = createButton(ButtonType.RightBumper);
+        m_buttonRT = createButton(ButtonType.RightTrigger);
+        m_buttonSelect = createButton(ButtonType.Select);
+        m_buttonStart = createButton(ButtonType.Start);
+
+        addShuffleboeardTab();
+    }
+
+    private IControllerMapping getControllerMapping() {
+        String joystickName = getControllerName().toLowerCase();
+        switch (joystickName) {
+            case "xbox controller":
+                return m_xboxControllerMapping;
+            default:
+                return m_logitechControllerMapping;
+        }
+    }
+
+    private String getControllerName() {
+        return m_joystick.getName();
+    }
+
+    public double getLeftX() {
+        return getControllerMapping().getLeftX(m_joystick);
+    }
+
+    public double getLeftY() {
+        return getControllerMapping().getLeftY(m_joystick);
+    }
+
+    public double getRightX() {
+        return getControllerMapping().getRightX(m_joystick);
+    }
+
+    public double getRightY() {
+        return getControllerMapping().getRightY(m_joystick);
+    }
+
+    public Button getButtonA() {
+        return m_buttonA;
+    }
+
+    public Button getButtonB() {
+        return m_buttonB;
+    }
+
+    public Button getButtonX() {
+        return m_buttonX;
+    }
+
+    public Button getButtonY() {
+        return m_buttonY;
+    }
+
+    public Button getButtonLB() {
+        return m_buttonLB;
+    }
+
+    public Button getButtonLT() {
+        return m_buttonLT;
+    }
+
+    public Button getButtonRB() {
+        return m_buttonRB;
+    }
+
+    public Button getButtonRT() {
+        return m_buttonRT;
+    }
+
+    public Button getButtonSelect() {
+        return m_buttonSelect;
+    }
+
+    public Button getButtonStart() {
+        return m_buttonStart;
+    }
+
+    private Button createButton(ButtonType buttonType) {
+        return new Button() {
+            public boolean get() {
+                return getControllerMapping().getButtonPressed(m_joystick, buttonType);
+            }
+        };
+    }
+
+    private void addShuffleboeardTab() {
+        ShuffleboardTab tab = Shuffleboard.getTab("Gamepad - " + m_controllerTitle);
```

Comment:
Added code to show the buttons on the shuffle board to easily debug problems. 
Posted a screenshot in the PR description.